dipy/denoise/denspeed.pyx

```diff
@@ -103,10 +108,9 @@ def _nlmeans_3d(double [:, :, ::1] arr, double [:, :, ::1] mask,
 
     # move the block
     with nogil, parallel():
-
-        for i in prange(B, I - B):
-            for j in range(B, J - B):
-                for k in range(B, K - B):
+        for i in prange(B + P, I - B - P):
```

Comment:
@Garyfallidis The bug for the cpu speed mismatch was pretty simple, I didn't use `prange` here which was giving that error. 
The times are, for the denoise_nlmeans example from doc/examples/ 
Original = 4.70s
Updated = 4.65s
so it's not much different. Also in the quality the updated one gives a much better denoised output (maybe too smooth at times), it also solves the bug reported in #1011 
I will commit those simple changes, I also think that this can be merged with the main branch, it's nearly done.